com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -1112,7 +1112,7 @@ internal static void InvokeRpc(RpcFrameQueueItem queueItem)
             var networkUpdateStage = queueItem.NetworkReader.ReadByteDirect();
             var networkMethodId = queueItem.NetworkReader.ReadUInt32Packed();
 
-            if (__ntable.ContainsKey(networkMethodId))
+            if (__ntable.TryGetValue(networkMethodId, out Action<NetworkBehaviour, NetworkSerializer, __RpcParams> action))
```

Comment:
Do you have to declare the type of the out value?  Might be easier to read without it here.

Reply:
Yes we have to declare the out value type. The reason why that's the case is for clarify. Else you wouldn't know the type of action when looking at the code.
The reason why this is ugly here is because we used the generic action type instead of defining our own short delegate type. I'd argue that we should rather fix the delegate and strongly type this. The fact that I had to rename this into "action" shows that it is not easy to tell what this type represents.

Reply:
controversial :) we definitely have more than "reduce dictionary lookups" changes in this PR. I'd like to ask for cutting this PR into pieces. also I'm happy to address this change specifically, even potentially own this change myself entirely. I wonder what IL both approaches would generate so that we could compare them side-by-side. I'm more concerned about the performance rather than syntax/eye-candy here since this piece of code would be in the hot-path.

Reply:
I have removed  this change and will introduce a separate PR or let you own this change.

I don't quite understand your performance concerns. TryGetValue will always be faster then running 2 dictionary lookups right? Without TryGetValue we have to run the expensive Dictionary.FindEntry method multiple times (once on ContainsKey and once for the [] indexer).